scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -152,3 +153,35 @@ def __iter__(self):
 
     def get_generated_keys(self) -> List[str]:
         return [self._name]
+
+
+class AttackPointIteratorRepeat(AttackPointIterator):
+    """
+    Attack point iterator class that iterates 
+    over a configuration a repeat amount of times.
+    """
+
+    def __init__(self,
+                 operation: str,
+                 configuration: Dict,
+                 repetitions: int = 0) -> None:
+        """Initialize the repeated iterate if repetitions is not present
+          or set to below 0 it will do an infinite loop."""
+        assert "repeat" == operation
+        if repetitions > 0:
+            self._repetitions = repetitions
```

Comment:
There is no unittest for length in the infinite case. There is no unittest testing that there are at least some (say two or 3 elements being iterated when `repetitions = 0`).

Reply:
Outdated comment: when `repetitions = -1`